pkg/reconciler/autoscaling/kpa/kpa_test.go

```diff
@@ -295,9 +296,21 @@ func TestReconcile(t *testing.T) {
 			metric(testNamespace, testRevision),
 			defaultDeployment,
 		},
+		WithReactors: []clientgotesting.ReactionFunc{
+			func(action clientgotesting.Action) (handled bool, ret runtime.Object, err error) {
+				if attempts != 0 || !action.Matches("update", "podautoscalers") {
+					return false, nil, nil
+				}
+				attempts++
```

Comment:
`attempts` can only be 0 or 1. Can we make this a boolean? Also, should we try to make each test case having their own `attempts`?

Reply:
addressed.